libc/newhdrgen/yaml_to_classes.py

```diff
@@ -103,16 +107,63 @@ def fill_public_api(header_str, h_def_content):
     return h_def_content.replace("%%public_api()", header_str, 1)
 
 
-def main(yaml_file, h_def_file, output_dir):
+def add_function_to_yaml(yaml_file, function_details):
```

Comment:
why is `function_details` not a function object?

Reply:
It was mainly a question of usability. CLI arguments are inherently string-based, so it would make sense to make the CLI a list of strings rather than some type of complex object.

Reply:
I agree that it makes more sense for the user to pass a list of strings, but that list of strings should be used to initialize the `function_details` object.